### PR TITLE
Feature/jn1 4263

### PR DESCRIPTION
**Pull request by CRMTravis**

### CODE REVIEW

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentController.js [hunk 1]
```diff
@@ -60,14 +60,7 @@
...
       //Async Functions
-      helper.supplementaryCardHolderInit(component);
-      helper.TDSRCalculationBefore(component);
-      helper.ASLCalculations(component);
-      helper.calculateCreditorLife(component);
-      helper.minimumPaymentCalculations(component);
-      helper.TDSRCalculationAfter(component);
-      helper.setCardType(component); //JN1-4049 :: Kirti R :: Calculate the credit type
-      helper.annualFeesCalcualtions(component);
+      helper.revolvingLoanCalculations(component);
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentController.js:63]: @CRMTravis is thought TDSR is also used in non-revolving loans, also this function should be guarded for credit or line of credit only.

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentController.js [hunk 2]
```diff
@@ -87,6 +80,10 @@
    * @param {*} helper
    */
   handleProductDetailsEvent: function (component, event, helper) {
+    const isAuto = checkProductFamily(component, "Auto");
+    const isUnsecured = checkProductFamily(component, "Unsecured");
+    const isLineOfCredit = checkProductFamily(component, "Line Of Credit");
+    const isCreditCard = checkProductFamily(component, "Credit Card");
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentController.js:86]: isLineOfCredit and isCreditCard are not been used, please remove them

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js [hunk 3]
```diff
@@ -176,10 +176,10 @@
     let values = [];
     let totalDebt = 0;
     let totalDebtAfter = 0;
-    const isAuto = this.checkProductFamily(component, "Auto");
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js:179]: this is required for these functions to operate

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js [hunk 4]
```diff
@@ -282,10 +282,10 @@
     let tdsrBefore = 0;
     let values = [];
     let container = component.get("v.ChildContainer");
-    const isAuto = this.checkProductFamily(component, "Auto");
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js:285]: this is required for these functions to operate

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js [hunk 5]
```diff
@@ -322,10 +322,10 @@
   TDSRCalculationAfter: function (component) {
     let container = component.get("v.ChildContainer");
     let tdsrAfter = 0;
-    const isAuto = this.checkProductFamily(component, "Auto");
-    const isLineOfCredit = this.checkProductFamily(component, "Line Of Credit");
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js:326]: this is required for these functions to operate

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js [hunk 6]
```diff
@@ -423,10 +423,10 @@
    * @return {Number} ltv
    */
   collateralTypeApplicable: function (component, container) {
-    const isAuto = this.checkProductFamily(component, "Auto");
-    const isLineOfCredit = this.checkProductFamily(component, "Line Of Credit");
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js:427]: this is required for these functions to operate

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js [hunk 7]
```diff
@@ -435,19 +435,6 @@
       return container.collateralType;
     }
   },
-  /**
-   * checks if the passed family is the selected product
-   * @param {*} component
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js:440]: why did you comment out checkProductFamily function?

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js [hunk 8]
```diff
@@ -748,10 +735,10 @@
     loanCalculationFields,
     loanCalculationProductFields
   ) {
-    const isAuto = this.checkProductFamily(component, "Auto");
-    const isLineOfCredit = this.checkProductFamily(component, "Line Of Credit");
-    const isUnsecured = this.checkProductFamily(component, "Unsecured");
-    const isCreditCard = this.checkProductFamily(component, "Credit Card");
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js:754]: this is required

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js [hunk 9]
```diff
@@ -748,10 +735,10 @@
...
+    const isAuto = checkProductFamily(component, "Auto");
+    const isLineOfCredit = checkProductFamily(component, "Line Of Credit");
+    const isUnsecured = checkProductFamily(component, "Unsecured");
+    const isCreditCard = checkProductFamily(component, "Credit Card");
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js:741]: this is required

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js [hunk 10]
```diff
@@ -955,5 +942,19 @@
...
     totalClosingCostFinancedJNCalculation(component);
     totalClosingCostPaidByApplicantCalculation(component);
+  },
+  /**
+   * all the revolving loan calculations
+   * @param {*} component
+   */
+  revolvingLoanCalculations: function (component, container) {
+    helper.supplementaryCardHolderInit(component);
+    helper.TDSRCalculationBefore(component);
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js:952]: replace helper with this

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js [hunk 11]
```diff
@@ -955,5 +942,19 @@
...
+    helper.ASLCalculations(component);
+    helper.calculateCreditorLife(component);
+    helper.minimumPaymentCalculations(component);
+    helper.TDSRCalculationAfter(component);
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js:956]: TDSR calculations apply to both revolving and non-revolving so remove it

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js (hunk 7 above)
Comment (RemarioCRM) [reply force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentHelper.js:440]: i saw where you placed it in the static resource, keep it there, ignore the other comments about this

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentController.js [hunk 12]
```diff
@@ -58,9 +58,15 @@
...
+      const isLineOfCredit = checkProductFamily(component, "Line Of Credit");
+      const isCreditCard = checkProductFamily(component, "Credit Card");
       noNotifyContainerChanges(component);
       //Async Functions
-      helper.revolvingLoanCalculations(component);
+      helper.TDSRCalculationBefore(component);
+      helper.TDSRCalculationAfter(component);
+      if (isLineOfCredit || isCreditCard) {
+        helper.revolvingLoanCalculations(component);
+      }
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentController.js:69]: call the helper.nonRevolvingLoanCalculations(component); here as well, remove it from the event handler

File: force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentController.js [hunk 13]
```diff
@@ -121,7 +122,9 @@
         helper.processingFeeCalculation(container, component)
       );
       //non revovling loan calculations
-      helper.nonRevolvingLoanCalculations(component, container);
+      if (isAuto || isUnsecured) {
+        helper.nonRevolvingLoanCalculations(component, container);
```
Comment (RemarioCRM) [inline force-app/main/default/aura/ProductDetailComponent/ProductDetailComponentController.js:126]: place in onChildContainerChange

File: force-app/main/default/staticresources/productDetailsGeneralFunctions.js [hunk 14]
```diff
@@ -71,15 +71,25 @@ window.calculateSavings = function (component, data) {
...
-  let total = calculateTotalLoanAmount(
-    [
-      "loanAmount",
-      "jnLifeCreditorPremium",
-      "processingFeesGCT",
-      "jngiMotorPremium"
-    ],
-    parentObj
-  );
+  const isAuto = this.checkProductFamily(component, "Auto");
```
Comment (RemarioCRM) [inline force-app/main/default/staticresources/productDetailsGeneralFunctions.js:74]: remove this

File: force-app/main/default/staticresources/productDetailsGeneralFunctions.js [hunk 15]
```diff
@@ -71,15 +71,25 @@ window.calculateSavings = function (component, data) {
...
+  const isUnsecured = this.checkProductFamily(component, "Unsecured");
```
Comment (RemarioCRM) [inline force-app/main/default/staticresources/productDetailsGeneralFunctions.js:75]: remove this

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/staticresources/productDetailsGeneralFunctions.js [hunk 16]
```diff
@@ -185,39 +218,74 @@ window.getFieldsToCalculate = function (parentObj) {
...
+      properties = ["stampDutyAuto", "legalFee", "nsipp"].concat(
+        fieldsTocalculate
+      );
+    }
+    total = calculateTotalClosingCost(properties, data);
+    return result;
+  } else if (isUnsecured) {
+    if (
+      component.get("v.estimatedStampDuty") != 0 &&
+      component.get("v.assignmentFee") != 0
```
Comment (RemarioCRM) [inline force-app/main/default/staticresources/productDetailsGeneralFunctions.js:246]: these two attributes are not on ProductDetails, i think it's on the ChildContainer, remember we not using the child components attributes directly @CRMTravis

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/staticresources/productDetailsGeneralFunctions.js (hunk 14 above)
Comment (CRMTravis) [reply force-app/main/default/staticresources/productDetailsGeneralFunctions.js:74]: Why?

File: force-app/main/default/staticresources/productDetailsGeneralFunctions.js (hunk 15 above)
Comment (CRMTravis) [reply force-app/main/default/staticresources/productDetailsGeneralFunctions.js:75]: Why?

**Review by RemarioCRM — APPROVED** [review]

**Pull request merged**